extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/ClassNames.java

```diff
@@ -465,4 +465,22 @@ private static DotName createConstant(String fqcn) {
             "io.quarkus.hibernate.orm.deployment.HibernateUserTypeProcessor");
 
     public static final DotName GRAAL_VM_FEATURES = createConstant("io.quarkus.hibernate.orm.deployment.GraalVMFeatures");
+
+    public static final List<DotName> SERVICE_PROVIDERS = List.of(
+            // Accessed in org.hibernate.query.sqm.internal.SqmCriteriaNodeBuilder.<init>
+            createConstant("org.hibernate.query.criteria.spi.CriteriaBuilderExtension"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.customized.QuarkusIntegratorServiceImpl.<init>
+            createConstant("org.hibernate.integrator.spi.Integrator"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.customized.QuarkusStrategySelectorBuilder.buildSelector
+            createConstant("org.hibernate.boot.registry.selector.StrategyRegistrationProvider"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.recording.RecordableBootstrap.applyServiceContributors
```

Comment:
`RecordableBootstrap` should only be used during static init... what was the exact stack trace?

Note that service contributors were explicitly disabled in the only place where they could legitimately be retrieved at runtime:

https://github.com/quarkusio/quarkus/blob/6daa2ea1952c0326afd232f453d378d01296e268/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/registry/PreconfiguredServiceRegistryBuilder.java#L110-L112